framework/aster-frame/src/mm/io.rs

```diff
@@ -160,28 +168,70 @@ impl_vmio_pointer!(&mut T, "(**self)");
 impl_vmio_pointer!(Box<T>, "(**self)");
 impl_vmio_pointer!(Arc<T>, "(**self)");
 
-/// VmReader is a reader for reading data from a contiguous range of memory.
+/// `VmReader` is a reader for reading data from a contiguous range of memory.
+///
+/// The memory range read by `VmReader` can be in either kernel space or user space.
+/// When the operating range is in kernel space, the memory within that range
+/// is guaranteed to be valid.
+/// When the operating range is in user space, it is ensured that the page table of
+/// the process creating the `VmReader` is active for the duration of `'a`.
 pub struct VmReader<'a> {
     cursor: *const u8,
     end: *const u8,
     phantom: PhantomData<&'a [u8]>,
+    /// Whether the memory scope of the `VmReader` is valid during the
+    /// entire period.
+    ensure_valid: bool,
```

Comment:
I think this boolean variable can and should be tracked at compile time rather than runtime.